nmi_mysql/nmi_mysql.py

```diff
@@ -1,194 +1,122 @@
+    Useful for raw queries and scripting
 """
 
+from sqlalchemy import create_engine
+
 import re
 import logging
-import pymysql.cursors
-from queue import Queue
-
-
-class DB(object):
-
-    def __init__(self, conf, autoconnect=False):
-        self.logger = logging.getLogger('database')
-        self.host = conf['host']
-        self.user = conf['user']
-        self.password = conf['password']
-        self.db_conn = conf['db']
-        self.port = int(conf['port'])
-        self.handle = None
-        self.connected = False
-
-        if autoconnect:
-            self.connect()
-
-    def __del__(self):
-        self.close()
 
-    def connect(self):
-
-        self.logger.info('Trying to connect to mysql database')
-        try:
-            con = pymysql.connect(host=self.host, user=self.user, password=self.password,
-                                  db=self.db_conn, port=self.port, charset='utf8mb4',
-                                  cursorclass=pymysql.cursors.DictCursor)
-
-        except Exception as err:
-            self.logger.error('Failed to connect to db')
-            self.logger.warn('Error:')
-            self.logger.info(err)
-            raise err
 
-        self.logger.info('Connection to mysql')
-        self.connected = True
-        self.handle = con
-        return True
+CONFIG_KEYS = ['host', 'user', 'password', 'db', 'port']
+MAX_POOL_SIZE = 10
 
-    def close(self):
-        try:
-            if self.connected:
-                self.handle.close()
-                self.connected = False
-                self.handle = None
 
-        except Exception as err:
-            self.logger.warn('Failed to close connection')
-            self.logger.warn(err)
-            raise err
-
-        return None
-
-    def query(self, _query, _params=None):
-        """
-            self.handle holds the connection
-            _query is the query
-            _params holds the variables need by the query
-        """
-
-        result = None
-        query = _query
-
-        if _params:
-            query = self.generate_query(_query, _params)
-
-        try:
-            with self.handle.cursor() as cursor:
-                cursor.execute(query, ())
-
-                if query.lower().strip().find('select') == 0:
-                    result = list(cursor.fetchall())
-
-                else:
-                    result = {
-                        'affected_rows': cursor.rowcount
-                    }
-
-        except Exception as err:
-            self.logger.warn(err)
-            raise err
-
-        self.handle.commit()
+class DB(object):
 
-        return result
+    def __init__(self, conf):
+        for c in CONFIG_KEYS:
+            if c not in conf:
+                raise ValueError('Invalid config object')
 
-    def generate_query(self, _query, _params):
+        self.logger = logging.getLogger('database')
+        self.charset = 'utf8mb4'
+        self.con = None
+
+        self.engine = create_engine(
```

Comment:
Is it assumed that it's the user's responsibility to catch the errors for the initialization?